code/modules/shuttle/shuttle_events/_shuttle_events.dm

```diff
@@ -129,7 +129,7 @@
 	. = pick_weight(spawning_list)
 	if(remove_from_list_when_spawned) //if we have this enabled, we decrease the pickweight by 1 till it runs out
 		spawning_list[.] -= 1
-		if(spawning_list[.] < 1)
+		if(spawning_list[.] < 0)
```

Comment:
Why below zero? Pickweight will never pick a value if it is set to 0, so -1 shouldn't occur

Reply:
Because `spawn_list.Remove(.)` removes the selected object (and deletes it?) before it's properly returned.

Reply:
Hmmm, very strange.  Testing it just now the previous `1` value now seems to work fine.  Maybe it was because I had set the list to be `spawning_list = list(/obj/singularity/shuttle_event)` instead of `spawning_list = list(/obj/singularity/shuttle_event = 1)` or playing with `self_destruct_when_empty`.

Reverted with https://github.com/tgstation/tgstation/pull/77188/commits/b69ed678202e8451985eaa81f8f211f7f10e28d9